src/diagnostics/MOM_diagnostics.F90

```diff
@@ -619,6 +620,22 @@ subroutine calculate_diagnostic_fields(u, v, h, uh, vh, tv, ADp, CDp, p_surf, &
       enddo
       if (CS%id_rhoinsitu > 0) call post_data(CS%id_rhoinsitu, Rcv, CS%diag)
     endif
+
+    if (CS%id_drho_dT > 0 .or. CS%id_drho_dS > 0) then
+      !$OMP parallel do default(none) shared(tv,Rcv,is,ie,js,je,nz,pressure_1d,h,GV)
+      do j=js,je
+        pressure_1d(:) = 0. ! Start at p=0 Pa at surface
+        do k=1,nz
+          pressure_1d(:) =  pressure_1d(:) + 0.5 * h(:,j,k) * GV%H_to_Pa ! Pressure in middle of layer k
+          ! To avoid storing more arrays, put drho_dT into Rcv, and drho_dS into work3d
+          call calculate_density_derivs(tv%T(:,j,k),tv%S(:,j,k),pressure_1d, &
```

Comment:
You need to import this function in module specification (L18-19).

(Was this able to build for you?)

Reply:
Aha, this was my error. I forgot to push my latest commit after making that change 🤦

Reply:
All good :tada: 
BTW do you think you are able to sort out the OpenMP error as well?  If not, then I can look into it.

Reply:
I don't think I have the knowledge to sort that. I'd be happy if you would talk me through it though.

Is it failing because of something in the changes that I've made? Or have I done something wrong in not merging my branch with dev/gfdl correctly in the first place?

Reply:
The OpenMP directive preceding the block (line starting with `!$OMP`) has an error because you have `default(none)` and haven't specified if `work_3d` is shared across threads or not.

The directive could be fixed by adding `work_3d` to the OpenMP directive.  But if you did not mean to add it, then it might be simpler to just remove it.

Reply:
Not knowing how OpenMP functions, I don't have a sense for whether it is necessary to have `work_3d` shared across threads. I was just copying the convention of the other density calculations in including this line.

I think, in fact, that the line as present in my PR was copied from a previous version of MOM_diagnostic.F90, as I see that the OpenMP directives for the other density variables (e.g. `rhoinsitu`) have changed. I wonder if this line should instead just be consistent with that for `rhoinsitu`? i.e. `!$OMP parallel do default(shared) private(pressure_1d)`

I will implement that just now and you can tell me if I'm missing something.

Reply:
That looks good, and also looks like it passed.  (Just waiting on the ARM tests...)
The regression fail is expected since this is a new diagnostic.  I'll push this to Gaea and if it passes then I will merge it.

Thanks for fixing that up!